### PR TITLE
Allow transformer to skip calls on x86

### PR DESCRIPTION
**Pull request by s1341**
Adds a test and a fix allowing transformer to skip calls on x86, inspired by https://github.com/frida/frida-gum/commit/10160db8588051e87cecc94902c59846116be43a

### CODE REVIEW

File: tests/core/arch-x86/stalker-x86.c [hunk 1]
```diff
@@ -930,6 +933,48 @@ store_xax (GumCpuContext * cpu_context,
...
+  const cs_insn * insn;
+
+  while (gum_stalker_iterator_next (iterator, &insn))
+  {
+    if (insn->address == GPOINTER_TO_SIZE (func_start + 5))
+      continue;
+
+    gum_stalker_iterator_keep (iterator);
+  }
+
```
Comment (oleavr) [inline tests/core/arch-x86/stalker-x86.c:975]:
```suggestion
```

File: tests/core/arch-x86/stalker-x86.c [hunk 2]
```diff
@@ -930,6 +933,48 @@ store_xax (GumCpuContext * cpu_context,
...
+TESTCASE (transformer_should_be_able_to_skip_call)
+{
+  guint8 code_template[] =
+  {
+    0xb8, 0x14, 0x05, 0x00, 0x00, //        mov    eax,0x514
+    0xe8, 0x01, 0x00, 0x00, 0x00, //        call   b <bump_number>
+    0xc3,                         //        ret
+                                  //  0000000b <bump_number>:
+    0x83, 0xc0, 0x25,             //        add    eax,0x25
+    0xc3                          //        ret
```
Comment (oleavr) [inline tests/core/arch-x86/stalker-x86.c:945]:
```suggestion
    0xb8, 0x14, 0x05, 0x00, 0x00, /* mov eax, 1300    */
    0xe8, 0x01, 0x00, 0x00, 0x00, /* call bump_number */
    0xc3,                         /* ret              */
    /* bump_number:                                   */
    0x83, 0xc0, 0x25,             /* add eax, 37      */
    0xc3,                         /* ret              */
```

**Review by oleavr — APPROVED** [review]
Yay, thanks! :raised_hands:

**Comment by yujack008** [comment]
@s1341  @oleavr I try but fail. I use Frida gum version is frida-gum-devkit-16.1.5-android-arm64.tar
```
static void transformer_callback(GumStalkerIterator * iterator,GumStalkerOutput * output, gpointer user_data){
    const cs_insn * insn;
    while (gum_stalker_iterator_next (iterator, &insn))
    {
        gboolean in_target = false;
        if(insn->address >= (g_il2cpp_base + 0x17A3270) && insn->address <= (g_il2cpp_base + 0x17A3788)){
            in_target = true;
        }
        if(in_target){
            DEBUG ("[*] %p\t%s\t%s", (gpointer)insn->address, insn->mnemonic, insn->op_str);
        }

        if(in_target && insn->id == ARM64_INS_BL){
            continue;
        }
        gum_stalker_iterator_keep (iterator);
    }
}
```
My Log:
11-17 22:35:05.824 21941 22038 E DiyHook : [*] 0x7b8c4ed75c     mov     x20, x1
11-17 22:35:05.824 21941 22038 E DiyHook : [*] 0x7b8c4ed760     mov     x1, xzr
11-17 22:35:05.824 21941 22038 E DiyHook : [*] 0x7b8c4ed764     mov     x19, x2
11-17 22:35:05.824 21941 22038 E DiyHook : [*] 0x7b8c4ed768     mov     x21, x0
11-17 22:35:05.824 21941 22038 E DiyHook : [*] 0x7b8c4ed76c     bl      #0x7b8be6ced0

I need to update Frida gum version ?

**Pull request merged**